File: Daily challenge/Reverse Prefix of Word.js

var reversePrefix = function(word, ch) 
{
    let rev = word;
    for (let i = 0; i < word.length; ++i) 
    {
        if (word[i] === ch) 
        {
            rev = word.slice(0, i + 1).split('').reverse().join('') + word.slice(i + 1);
            break;
        }
    }
    return rev;    
};

File: Daily challenge/Reverse Prefix of Word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) 
    {
        string rev = word;
        for (int i = 0; i < word.length(); ++i) {
            if (word[i] == ch) {
                rev = word.substr(0, i + 1);
                std::reverse(rev.begin(), rev.end());
                rev += word.substr(i + 1);
                break;
            }
        }
        return rev;    
    }
};